### ISSUE TITLE
Listen to channel state changes on Android

### ISSUE THREAD

**Issue by Gericop** [grpc/grpc-java#2484]
It's more of a question than an issue. I can't find any methods to listen to connection state changes. Is there any way to do this on the client side? I'm using protobuf-lite on Android. It'd be nice to listen to channel state changes, e.g. if the Internet connection is lost, the channel should signal the error. I have a persistent connection (full-duplex) to the server with a `StreamObserver`, but neither the `onError(...)` nor `onCompleted()` gets called if I just turn off the WiFi to test it.

**TL;DR**: How can I know on Android whether the channel is still operational (so the connection is not down)?

**Comment by ejona86** [comment]
Your `StreamObserver` solution should work. I don't know why it is failing, but it should be looked into.

We're adding `getState()`/`notifyWhenStateChanged` APIs to `ManagedChannel` in 1.1. These are currently only implemented for some load balancing logic, but by 1.1 they should be implemented for normal use cases as well (tracked in #2292).

**Comment by dapengzhang0** [comment]
A generic Chanel state API is implemented in 1.6.1.
